content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java

```diff
@@ -4250,6 +4250,41 @@ public int compare(Group g0, Group g1) {
 					// something failed, group filter will be hidden
 				}
 			}
+
+			//SAK-11647 - Group-aware dropboxes
+			try
+			{
+				String currentUser = SessionManager.getCurrentSessionUserId();
+				Site site = SiteService.getSite(ToolManager.getCurrentPlacement().getContext());
+				
+				if ((!AuthzGroupService.isAllowed(currentUser,ContentHostingService.AUTH_DROPBOX_MAINTAIN,SiteService.siteReference(site.getId())))
+						&&(AuthzGroupService.isAllowed(currentUser,ContentHostingService.AUTH_DROPBOX_GROUPS,SiteService.siteReference(site.getId()))))
+				{
+					context.put("dropboxGroupPermission_enabled",Boolean.TRUE);
+					
+					List<Group> site_groups = new ArrayList<Group>();
+					
+					Set allGroupsUsers = new TreeSet<String>();
+					
+					site_groups.addAll(site.getGroupsWithMember(currentUser));
+					if (site_groups.size()>0)
+					{
+						for (Group g : site_groups)
+						{
+							allGroupsUsers.addAll(g.getUsers());
+						}
+					}
+					context.put("dropboxGroupPermission_allGroupsUsers",allGroupsUsers);
+				}
+				else
+				{
+					context.put("dropboxGroupPermission_enabled",Boolean.FALSE);
+				}
+			}
+			catch (IdUnusedException e)
+			{
+				logger.debug("DropboxGroupPermission error: "+e.toString());
```

Comment:
Are you expecting this error to happen? If not then it's probably worth raising it to a warn level.